packages/arb-token-bridge-ui/src/components/MainContent/MainContent.tsx

```diff
@@ -29,14 +31,31 @@ export const motionDivProps = {
 export function MainContent() {
   const { address } = useAccount()
   const { closeTransactionHistoryPanel } = useAppContextActions()
-  const transactionHistoryProps = useTransactionHistory(address)
+  const transactionHistoryProps = useTransactionHistory(address, true)
```

Comment:
It's not great. But we need to ensure that only one instance of the hook handles pagination, otherwise `SWRInfinite` revalidates each time `setPage` is updated.

I tried using mutex instead but it would bring even more complexity because we need to ensure that acquired lock applies to the specific hook instance and that only this instance is calling every useEffect update. That required storing extra data which brought more complexity than just using the flag here.